src/compiler/utilities.ts

```diff
@@ -1913,6 +1914,12 @@ namespace ts {
         return token >= SyntaxKind.FirstAssignment && token <= SyntaxKind.LastAssignment;
     }
 
+    export function isClassExtendsExpressionWithTypeArguments(node: Node): boolean {

```

Comment:
The kind can be checked before calling the function.

If you keep the check you can call it `isExpressionWithTypeArgumentsInClassExtendsClause`.